acceptance/tests/utf8/utf8-in-file-resource.rb

```diff
@@ -0,0 +1,55 @@
+test_name 'utf-8 characters in resource title and param values' do
+  confine :except, :platform => [ 'windows', 'ubuntu-16']   # PUP-6983
+  utf8chars = "€‰ㄘ万竹ÜÖ"
+  agents.each do |agent|
+    agent_file = agent.tmpfile("file" + utf8chars) 
+    # remove this file, so puppet can create it and not merely correct
+    # its drift.
+    on(agent, "rm -rf #{agent_file}")
+
+    manifest =
+<<PP
+
+file { "#{agent_file}" :
+  ensure => file,
+  mode => "0644",
+  content => "This is the file content. file #{utf8chars} 
+",
+}
+
+PP
+
+    step "Apply manifest" do
+      apply_manifest_on(
+        agent,
+        manifest,
+        {:acceptable_exit_codes => [0, 2], :catch_failures => true, }
+      )
+      result = on(agent, "cat #{agent_file}")
+      assert_equal(result.exit_code, 0)
+      assert_match(
+        /#{utf8chars}/,
+        result.stdout,
+        "result stdout did not contain"
+      )
+    end
+
+    step "Drift correction" do
+      on(agent, "> #{agent_file}")
```

Comment:
Just for posterity, this fails on cisco nx, because to escape the cisco management interface and get to bash, commands are prefixed with `ip netns exec management`.

The redirection here is interpreted as redirecting the output of a command `ip netns exec management` to the file
(but there isn't a command to run to redirect the output of). The output looked like:
```
01:29:25       * Drift correction
01:29:25         
01:29:25         chtifw7vi39jwhs.delivery.puppetlabs.net (cisconx-64-1) 09:29:24$ source /etc/profile; sudo sh -c " ip netns exec management > /tmp/file€‰ㄘ万竹ÜÖ.cA1zbl "
01:29:25           No cmd specified
01:29:25         
01:29:25         chtifw7vi39jwhs.delivery.puppetlabs.net (cisconx-64-1) executed in 0.08 seconds
01:29:25         Exited: 1
```